meilisearch-http/src/snapshot.rs

```diff
@@ -0,0 +1,134 @@
+use crate::Data;
+use crate::error::Error;
+
+use flate2::Compression;
+use flate2::read::GzDecoder;
+use flate2::write::GzEncoder;
+use log::{error, warn};
+use std::fs::{create_dir_all, File};
+use std::io;
+use std::path::{PathBuf, Path};
+use std::thread;
+use std::time::{Duration};
+use tar::{Builder, Archive};
+use tempfile::TempDir;
+
+fn pack(src: &Path, dest: &Path) -> io::Result<()> {
+    let f = File::create(dest)?;
+    let mut gz_encoder = GzEncoder::new(f, Compression::default());
+    {
+        let mut tar_encoder = Builder::new(&mut gz_encoder);
+        tar_encoder.append_dir_all(".", src)?;
+        tar_encoder.finish()?;
+    }
+    gz_encoder.finish()?;
+
+    Ok(())
```

Comment:
Notice the `&mut gz_encoder` replaced by `gz_encoder` here.

```suggestion
    let f = File::create(dest)?;
    let gz_encoder = GzEncoder::new(f, Compression::default());
    let mut tar_encoder = Builder::new(gz_encoder);
    tar_encoder.append_dir_all(".", src)?;
    let gz_encoder = tar_encoder.into_inner()?;
    gz_encoder.finish()?;
    Ok(())
```